src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,21 +171,21 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager();
```

Comment:
It would be easier to understand if you just wrote 
var effectiveConfigurationManager = IsV2Authority ? ConfigurationManagerV2 : ConfigurationManagerV1;

Reply:
changed to check iss from token rahter than the isV2Authority